File: analyze_desktop_db.js
const sqlite3 = require('sqlite3').verbose();
const { Pool } = require('pg');
const fs = require('fs');

const DB_PATH = 'C:/Users/saul_/AppData/Local/Packages/6a727d9d-d40f-407d-a7b7-655ca0f8161b_pkzpc8njrvjtr/LocalState/SYATortillerias.db3';

const pool = new Pool({
  host: 'dpg-d3i8dv3e5dus738tm5rg-a.oregon-postgres.render.com',
  user: 'sya_admin',
  password: 'qJ1haIaPp7m7OFMyicWSplPlGoNL1GpF',
  database: 'sya_db_oe4v',
  port: 5432,
  ssl: { rejectUnauthorized: false }
});

async function analyzeDatabase() {
  return new Promise((resolve, reject) => {
    const db = new sqlite3.Database(DB_PATH, sqlite3.OPEN_READONLY, (err) => {
      if (err) {
        reject(err);
        return;
      }

      console.log('📊 Analizando base de datos SQLite del Desktop...\n');

      // Obtener lista de tablas
      db.all("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name", async (err, tables) => {
        if (err) {
          reject(err);
          return;
        }

        const analysis = {};

        for (const table of tables) {
          const tableName = table.name;

          // Obtener estructura de tabla
          const tableInfo = await new Promise((res, rej) => {
            db.all(`PRAGMA table_info(${tableName})`, (err, columns) => {
              if (err) rej(err);
              else res(columns);
            });
          });

          // Obtener conteo de registros
          const count = await new Promise((res, rej) => {
            db.get(`SELECT COUNT(*) as count FROM ${tableName}`, (err, result) => {
              if (err) rej(err);
              else res(result.count);
            });
          });

          analysis[tableName] = {
            columns: tableInfo,
            recordCount: count
          };
        }

        db.close();

        // Verificar qué tablas existen en PostgreSQL
        console.log('═══════════════════════════════════════════════════════');
        console.log('COMPARACIÓN DESKTOP (SQLite) vs CLOUD (PostgreSQL)');
        console.log('═══════════════════════════════════════════════════════\n');

        const { rows: pgTables } = await pool.query(`
          SELECT table_name FROM information_schema.tables
          WHERE table_schema = 'public' AND table_type = 'BASE TABLE'
        `);

        const pgTableNames = new Set(pgTables.map(t => t.table_name));

        const criticalTables = [
          'Shifts',
          'CashCuts',
          'Expenses',
          'ExpenseCategories',
          'Purchases',
          'DeliveryAssignments',
          'Sales',
          'SaleItems',
          'Products',
          'Clientes',
          'Employees',
          'GuardianEvents'
        ];

        for (const tableName of criticalTables) {
          const lowerName = tableName.toLowerCase();
          const existsInPg = pgTableNames.has(lowerName);
          const existsInSqlite = analysis[tableName] !== undefined;

          console.log(`\n📋 Tabla: ${tableName}`);
          console.log(`   Desktop (SQLite): ${existsInSqlite ? '✅ SÍ' : '❌ NO'} ${existsInSqlite ? `(${analysis[tableName].recordCount} registros)` : ''}`);
          console.log(`   Cloud (PostgreSQL): ${existsInPg ? '✅ SÍ' : '❌ NO'}`);

          if (existsInSqlite) {
            console.log(`   Columnas en Desktop:`);
            analysis[tableName].columns.forEach(col => {
              console.log(`      - ${col.name.padEnd(30)} ${col.type.padEnd(15)} ${col.notnull ? 'NOT NULL' : 'NULL'}`);
            });

            if (existsInPg) {
              // Comparar columnas
              const { rows: pgColumns } = await pool.query(`
                SELECT column_name, data_type, is_nullable
                FROM information_schema.columns
                WHERE table_name = $1
                ORDER BY ordinal_position
              `, [lowerName]);

              const pgColNames = new Set(pgColumns.map(c => c.column_name));
              const sqliteColNames = new Set(analysis[tableName].columns.map(c => c.name.toLowerCase()));

              const missing = analysis[tableName].columns
                .filter(c => !pgColNames.has(c.name.toLowerCase()))
                .map(c => c.name);

              if (missing.length > 0) {
                console.log(`   ⚠️  Columnas FALTANTES en PostgreSQL:`);
                missing.forEach(col => console.log(`      ❌ ${col}`));
              }
            }
          }
        }

        // Guardar análisis completo
        fs.writeFileSync(
          'C:/SYA/sya-socketio-server/desktop_db_analysis.json',
          JSON.stringify(analysis, null, 2)
        );

        console.log('\n✅ Análisis completado y guardado en desktop_db_analysis.json\n');

        pool.end();
        resolve(analysis);
      });
    });
  });
}

analyzeDatabase().catch(error => {
  console.error('❌ Error:', error);
  pool.end();
  process.exit(1);
});

File: analyze_desktop_db.py
import sqlite3
import json

DB_PATH = r'C:\Users\saul_\AppData\Local\Packages\6a727d9d-d40f-407d-a7b7-655ca0f8161b_pkzpc8njrvjtr\LocalState\SYATortillerias.db3'

def analyze_database():
    conn = sqlite3.connect(DB_PATH)
    cursor = conn.cursor()

    print('📊 Analizando base de datos SQLite del Desktop...\n')

    # Obtener todas las tablas
    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' ORDER BY name")
    tables = [row[0] for row in cursor.fetchall()]

    analysis = {}

    for table_name in tables:
        # Obtener estructura
        cursor.execute(f'PRAGMA table_info({table_name})')
        columns = cursor.execute(f'PRAGMA table_info({table_name})').fetchall()

        # Obtener conteo
        cursor.execute(f'SELECT COUNT(*) FROM {table_name}')
        count = cursor.fetchone()[0]

        analysis[table_name] = {
            'columns': [
                {
                    'cid': col[0],
                    'name': col[1],
                    'type': col[2],
                    'notnull': bool(col[3]),
                    'default_value': col[4],
                    'pk': bool(col[5])
                }
                for col in columns
            ],
            'recordCount': count
        }

    conn.close()

    # Imprimir análisis
    print('═══════════════════════════════════════════════════════')
    print('TABLAS CRÍTICAS EN DESKTOP (SQLite)')
    print('═══════════════════════════════════════════════════════\n')

    critical_tables = [
        'Shifts',
        'CashCuts',
        'Expenses',
        'ExpenseCategories',
        'Purchases',
        'PurchaseItems',
        'DeliveryAssignments',
        'Sales',
        'SaleItems',
        'Products',
        'ProductCategories',
        'Clientes',
        'Employees',
        'GuardianEvents',
        'CashDrawerMovements',
        'DepositsWithdrawals'
    ]

    for table_name in critical_tables:
        if table_name in analysis:
            table_info = analysis[table_name]
            print(f'\n📋 Tabla: {table_name}')
            print(f'   Registros: {table_info["recordCount"]}')
            print(f'   Columnas:')
            for col in table_info['columns']:
                pk = ' [PK]' if col['pk'] else ''
                notnull = ' NOT NULL' if col['notnull'] else ''
                print(f'      - {col["name"]:<30} {col["type"]:<15}{notnull}{pk}')
        else:
            print(f'\n❌ Tabla: {table_name} - NO EXISTE en Desktop')

    # Guardar JSON
    with open('C:/SYA/sya-socketio-server/desktop_db_analysis.json', 'w') as f:
        json.dump(analysis, f, indent=2)

    print('\n✅ Análisis guardado en desktop_db_analysis.json\n')

    return analysis

if __name__ == '__main__':
    analyze_database()
